src/ExpressionEvaluator/Core/Test/ExpressionCompiler/MockSymUnmanaged.cs

```diff
@@ -331,6 +338,44 @@ int ISymUnmanagedNamespace.GetVariables(int cVars, out int pcVars, ISymUnmanaged
         }
     }
 
+    internal delegate int GetSignatureDelegate(int bufferLength, out int count, byte[] signature);
+
+    internal sealed class MockSymUnmanagedConstant : ISymUnmanagedConstant
+    {
+        private readonly string _name;
+        private readonly object _value;
+        private readonly GetSignatureDelegate _getSignature;
+
+        public MockSymUnmanagedConstant(string name, object value, GetSignatureDelegate getSignature)
+        {
+            _name = name;
+            _value = value;
+            _getSignature = getSignature;
+        }
+
+        int ISymUnmanagedConstant.GetName(int bufferLength, out int count, char[] name)
+        {
+            count = _name.Length + 1; // + 1 for null terminator
+            Debug.Assert((bufferLength == 0) || (bufferLength == count));
+            for (int i = 0; i < bufferLength - 1; i++)
+            {
+                name[i] = _name[i];
+            }
```

Comment:
If I do that, I end up having to null check ```name```, and it ends up being the same num lines of code, so I'm just going to leave it.